Documentation/ContinousIntegration.md

```diff
@@ -0,0 +1,27 @@
+# Building Swift Packages or Apps that Use Them in Continuous Integration Workflows
+
+Build Swift packages with an existing continuous integration setup and prepare apps that consume package dependencies within an existing CI pipeline.
+
+## Overview
+
+*Continuous integration* (*CI*) is the process of automating and streamlining the building, analyzing, testing, archiving, and publishing of your apps to ensure that they're always in a releasable state.
+Most projects that contain or depend on Swift packages don't require additional configuration.
+
+## Use the Expected Version of a Package Dependency
+
+To ensure the *CI* workflow’s reliability, make sure it uses the appropriate version of package dependencies.
+SwiftPM records the result of dependency resolution in `Package.resolved` (at the top-level of the package) and it's used when performing dependency resolution (rather than having SwiftPM searching the latest eligible version of each package).  Running `swift package update` updates all dependencies to the latest eligible versions and updates the `Package.resolved`.  You can commit `Package.resolved` to your *Git* repository to ensure it’s always up-to-date on the *CI* environment to prevent the *CI* from building your project with unexpected versions of package dependencies.  Otherwise you can choose to add `Package.resolved` file to `.gitignore` file and have `swift package resolve` command in charge of resolving the dependencies (`swift package resolve` is invoked by most SwiftPM commands).
+
+## Provide Credentials
+
+To resolve package dependencies that require authentication, or private packages, you need to provide credentials to your CI setup.
+Use SSH–based Git URLs for your packages and configure your SSH credentials. Set up your known_hosts file in the ~/.ssh directory of the user that runs your CI tasks. SwiftPM honors your SSH configuration - there's no additional setup required.
+If your SSH keys are password-protected, add them to the SSH agent before invoking SwiftPM by modifying the SSH configuration file.
+
+CI services like [Jenkins](https://www.jenkins.io/doc/book/using/using-credentials), [Github Action](https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow), [TravisCI](https://docs.travis-ci.com/user/private-dependencies), [CircleCI](https://circleci.com/docs/2.0/gh-bb-integration/#security), are providing ways to set up credentials.
```

Comment:
```suggestion
CI services like [Jenkins](https://www.jenkins.io/doc/book/using/using-credentials), [Github Action](https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow), [TravisCI](https://docs.travis-ci.com/user/private-dependencies), [CircleCI](https://circleci.com/docs/2.0/gh-bb-integration/#security) provide ways to set up SSH keys or other techniques to access private repositories. Since SwiftPM uses git to clone the repositories there's no additional setup required on SwiftPM side, and SwiftPM will honor the machine's SSH and git configuration.
```